### ISSUE TITLE
Latest dev docs

### ISSUE THREAD

**Issue by cdeil** [holoviz/panel#4544]
At https://pyviz-dev.github.io/panel I currently see the v0.14.4rc2 docs.

Shouldn't this be the URL for the latest docs for the `main` branch?

**Comment by philippjfr** [comment]
Indeed, the CI runner was running out of memory for the longest time. After switching to MacOS runner it seems to now be working again. I was hoping to finish off https://github.com/holoviz/panel/pull/4541 to be able to provide sensible developer instructions but I am holding off until @maximlt is back and can evaluate it (there's also some work left to clean that up).
